content/articles/nextjs-dnd-file-upload/index.md

```diff
@@ -0,0 +1,427 @@
+
+The `index.js` file:
+
+```js
+import React from "react";
+import Head from "next/head";
+import DropZone from "../components/DropZone";
+import styles from "../styles/Home.module.css";
+
+export default function Home() {
+  return (
+    <div className={styles.container}>
+      <Head>
+        <title>Drag And Drop File Upload</title>
+        <meta name="description" content="Nextjs drag and drop file upload" />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+
+      <main className={styles.main}>
+        <h1 className={styles.title}>Drag And Drop File Upload</h1>
+        <DropZone />
+      </main>
+
+      <footer className={styles.footer}>
+        <div>{new Date().getFullYear()}</div>
+      </footer>
+    </div>
+  );
+}
+```
+
+The index.js file will be used to render the application. It imports the `DropZone` component from the `components` directory. It uses the `Home.module.css` file to style the page. 
+
+The logic to handle and manage the state (the selected files) will be contained within the `index.js` file.
+
+Here is the UI of the application up to this point:
+
+![The dropzone ](/content/articles/nextjs-dnd-file-upload/dropzone.png)
+
+### Managing state
+To keep track of the dropped files, you will need to manage the state of the application. We will keep track of the following states:
+- `inDropZone` - A boolean value, will be set to `true` when the user drags a file over the dropzone region.
+- `fileList` - An array of files (file objects) that have been selected.
+
+The app state will depend on the previously selected files (previous state) and makes use of the `useReducer` hook to manage state changes. The hook takes in a reducer function and an initial state. 
+
+The reducer function will be used to update the state i.e `(state, action) => newState` . To read more about the useReducer hook click [here](https://reactjs.org/docs/hooks-reference.html#usereducer).
+
+In the `index.js` file before the return, add the following code:
+
+```js
+...
+ // reducer function to handle state changes
+  const reducer = (state, action) => {
+    switch (action.type) {
+      case "SET_IN_DROP_ZONE":
+        return { ...state, inDropZone: action.inDropZone };
+      case "ADD_FILE_TO_LIST":
+        return { ...state, fileList: state.fileList.concat(action.files) };
+      default:
+        return state;
+    }
+  };
+
+  // destructuring state and dispatch, initializing fileList to empty array
+  const [data, dispatch] = useReducer(reducer, {
+    inDropZone: false,
+    fileList: [],
+  });
+...
+
+```
+
+The `useReducer` hook takes in a reducer function and the initial state and returns the current state and a dispatch function. The dispatch function will be used to update the state.
```

Comment:
You've used the word `and` several times in one sentence making it hard to read. Kindly rephrase this sentence